content/blog/2024-12-28-robius-retrospective-2024.md

```diff
@@ -0,0 +1,356 @@
++++
+title = "Project Robius in 2024: one year of Rust App Dev"
+description = ""
+[extra]
+author = "Kevin Boos"
+twitter = ""
+mastodon = ""
+github = "kevinboos"
++++
+*Author: [Kevin Boos](https://github.com/kevinaboos). Published December 28th, 2024.*
+
+This past year marked the first year of work on **Project Robius**: an open-source decentralized endeavor to enable developers to write immersive, fully-featured apps in pure Rust that work seamlessly across all major platforms.
+
+> 🛈 Project Robius in a nutshell
+>
+> While GUIs are the foundation of an app, there is more to developing a modern featureful app than just drawing a UI.
+> Project Robius aims to fill in the gaps in the Rust app dev ecosystem by focusing on everything *except*  the UI, such as abstractions for platform features & OS services, build & packaging tooling, and more.
+> We leave UI work to the experts behind the many excellent Rust UI toolkits under active development.
+
+In this post, we'll take a look back on what we've accomplished so far to make the world of Rust App Dev a little better:
+1. The crate abstractions we've published for accessing platform-provided features from Rust code
+2. The major apps we've built using [Makepad] + Robius together
+3. The contributions we've made to existing open-source projects in the App Dev space
+4. The connections we've fostered throughout the Rust community
+
+We'll also take a deeper look at **Robrix**,  a multi-platform [Matrix](https://matrix.org/) chat client written from scratch in Rust using the [Makepad UI toolkit] and Robius components.
+
+## 1. Robius crates for platform feature abstractions
+
+As our primary objective, we have published several crates intended to be used directly by app devs to access a given platform feature or OS service from their app. The main goal here is for each crate to offer a safe, platform-agnostic abstraction, such that the app dev need not worry about writing any platform-specific code or dealing with each platforms' idiosyncracies.
+
+We began working on these crates in late Spring of 2024; development has been a bit intermittent but we are ramping up for the coming year. While we expect to publish many more in the future, here is the current list:
```

Comment:
The sentence "development has been a bit intermittent but we are ramping up for the coming year" could be more confident:
"While development started gradually, we are significantly ramping up our efforts for the coming year."

Reply:
great point, thanks